mfhdf/libsrc/xdrposix.c

```diff
@@ -260,23 +260,16 @@ static struct xdr_ops xdrposix_ops = {
     xdrposix_getpos,   /* get offset in the stream */
     xdrposix_setpos,   /* set offset in the stream */
     xdrposix_inline,   /* prime stream for inline macros */
-#if (defined __sun && defined _LP64) || defined __x86_64__ || defined __powerpc64__
-    xdrposix_destroy, /* destroy stream */
-#if !(defined __x86_64__) && !(defined __powerpc64__) ||                                                     \
-    (defined __sun && defined _LP64) /* i.e. we are on SUN/Intel in 64-bit mode */
-    NULL,                            /* no xdr_control function defined */
-#endif
-    /* Solaris 64-bit (arch=v9 and arch=amd64) has 64 bits long and 32 bits int. */
-    /* It defines the two extra entries for get/put int. here */
+    xdrposix_destroy,  /* destroy stream */
+    NULL,              /* no xdr_control function defined */
+#if defined(__sun) && defined(_LP64)
+    /* Solaris 64-bit (arch=v9 and arch=amd64) differentiates between
+     * 32-bit integers and 64-bit longs via two extra callbacks for
+     * int here. These are not present on other XDR implementations
+     * like TI-RPC.
+     */
     xdrposix_getint, /* deserialize a 32-bit int */
     xdrposix_putint  /* serialize a 32-bit int */
-#else
-#ifdef AIX5L64
-    xdrposix_destroy,  NULL,           NULL,
-    xdrposix_getint,   xdrposix_putint
-#else  /*AIX5L64 */
-    xdrposix_destroy /* destroy stream */
-#endif /*AIX5L64 */
```

Comment:
Until someone gives me access to an AIX machine, we don't explicitly support AIX anymore.

Reply:
It wouldn't be speedy, but maybe AIX on QEMU?

https://kwakousys.wordpress.com/2020/09/06/run-aix-7-2-on-x86-with-qemu/

Reply:
I've thought about using QEMU, mainly for emulating a BE machine if our existing ones die. I haven't thought about the remaining non-Linux, non-BSD/MacOS UNIX flavors, mainly because they've historically had expensive licenses for their compilers.

Reply:
"You can create a free account on IBM but you may have to provide details on an existing serial number and Power Series machine model prior to downloading."

Not sure where I would get something like that